velero-plugins/build/restore.go

```diff
@@ -86,7 +86,6 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	// if it is not sourceBuildStrategyType
 	build.Spec.Strategy.SourceStrategy.From.Name = newName
 	for _, trigger := range build.Spec.TriggeredBy {
-		trigger.ImageChangeBuild.FromRef.Name = newName
```

Comment:
Sounds like we need to check the Kind here to see if we need to swap this. I will test more but I found that using nginx on 3.10 and restoring to 4.0 I needed to swap this ref.

Reply:
Ignore :)